src/nft/components/sell/modal/utils.ts

```diff
@@ -0,0 +1,305 @@
+    callback: () =>
+      approveCollectionRow(
+        collectionRow,
+        collectionsRequiringApproval,
+        setCollectionsRequiringApproval,
+        signer,
+        looksRareAddress,
+        pauseAllRows
+      ),
+  })
+  const { marketplace, collectionAddress } = collectionRow
+  const spender =
+    marketplace.name === 'OpenSea'
+      ? OPENSEA_CROSS_CHAIN_CONDUIT
+      : marketplace.name === 'Rarible'
+      ? process.env.REACT_APP_LOOKSRARE_MARKETPLACE_CONTRACT
+      : marketplace.name === 'X2Y2'
+      ? process.env.REACT_APP_X2Y2_TRANSFER_CONTRACT
+      : looksRareAddress
+  await approveCollection(spender ?? '', collectionAddress, signer, (newStatus: ListingStatus) =>
+    updateStatus({
+      listing: collectionRow,
+      newStatus,
+      rows: collectionsRequiringApproval,
+      setRows: setCollectionsRequiringApproval as Dispatch<AssetRow[]>,
+    })
+  )
+  if (collectionRow.status === ListingStatus.REJECTED || collectionRow.status === ListingStatus.FAILED) pauseAllRows()
+}
+
+export async function signListingRow(
+  listing: ListingRow,
+  listings: ListingRow[],
+  setListings: Dispatch<ListingRow[]>,
+  signer: JsonRpcSigner,
+  provider: Web3Provider,
+  getLooksRareNonce: () => number,
+  setLooksRareNonce: (nonce: number) => void,
+  pauseAllRows: () => void
+) {
+  const looksRareNonce = getLooksRareNonce()
+  updateStatus({
+    listing,
+    newStatus: ListingStatus.SIGNING,
+    rows: listings,
+    setRows: setListings as Dispatch<AssetRow[]>,
+    callback: () => {
+      return signListingRow(
+        listing,
+        listings,
+        setListings,
+        signer,
+        provider,
+        getLooksRareNonce,
+        setLooksRareNonce,
+        pauseAllRows
+      )
+    },
+  })
+  const { asset, marketplace } = listing
+  const res = await signListing(marketplace, asset, signer, provider, looksRareNonce, (newStatus: ListingStatus) =>
+    updateStatus({
+      listing,
+      newStatus,
+      rows: listings,
+      setRows: setListings as Dispatch<AssetRow[]>,
+    })
+  )
+  if (listing.status === ListingStatus.REJECTED) pauseAllRows()
+  else {
+    res && listing.marketplace.name === 'LooksRare' && setLooksRareNonce(looksRareNonce + 1)
+    const newStatus = res ? ListingStatus.APPROVED : ListingStatus.FAILED
+    updateStatus({
+      listing,
+      newStatus,
+      rows: listings,
+      setRows: setListings as Dispatch<AssetRow[]>,
+    })
+  }
+}
+
+export const getTotalEthValue = (sellAssets: WalletAsset[]) => {
+  const total = sellAssets.reduce((total, asset: WalletAsset) => {
+    if (asset.newListings?.length) {
+      const maxListing = asset.newListings.reduce((a, b) =>
+        parseFloat(a.price ?? '') > parseFloat(b.price ?? '') ? a : b
+      )
+      return (
+        total +
+        parseFloat(maxListing.price ?? '') -
+        parseFloat(maxListing.price ?? '') * (maxListing.marketplace.fee / 100 + asset.creatorPercentage)
```

Comment:
should all these `parseFloat` functions be `?? ''` or would it make more sense to do `?? 0`

Reply:
parseFloat takes a string so it just needed to be '0'